src/coreclr/jit/hwintrinsic.h

```diff
@@ -866,6 +866,11 @@ struct HWIntrinsic final
             {
                 baseType = node->TypeGet();
             }
+
+            if (category == HW_Category_Scalar)
```

Comment:
What about the `HW_Category_Special` category? Likely the above three cases that get a type via `GenTree::TypeGet` should be changed to use `genActualType(tree)` instead.

Reply:
I was trying to make a targeted change, and hadn't fully investigated all the processing that "Special" nodes have. It turns out that this class is only used on arm64, and the only "special" instruction is "Yield", which has no arguments, and for which the "baseType" is ignored. So I could just make the `genActualType` unconditional as you suggest.